### DISCUSSION TITLE
Retrieve storage primary key

### DISCUSSION THREAD

**Discussion by sitoder** [Azure/ResourceModules#2655] [Q&A]
In my main.bicep I create a storage account:

`module r_storage 'br/modules:microsoft.storage.storageaccounts:latest'={
  name:p_storageName
  params:{
    name:p_storageName
    location:p_location
    storageAccountSku:'Standard_LRS'
    storageAccountKind:'StorageV2'
    storageAccountAccessTier:'Hot'
    allowBlobPublicAccess:false
    minimumTlsVersion:'TLS1_2'
    fileServices:{
      shares:[
        {
          name:p_storageMountName
          shareQuota:10
        }
      ]
    }
  }
}`

Afterwards in the same main, i need to retrieve the storage account's primary key in order to create a container app environment mount point.
`resource r_storageMount 'Microsoft.App/managedEnvironments/storages@2022-06-01-preview' = {
  name: p_storageMountName
  parent: r_cae
  dependsOn:[
    r_storage
  ]
  properties: {
    azureFile: {
      accessMode: 'ReadWrite'
      accountKey: ??????????????????
      accountName: r_storage.outputs.name
      shareName: p_storageMountName
    }
  }
}`
Since the storage module does not output the key for obvious reasons, I am not really sure how i should use the listKeys function to get that. 
I'd rather not to use a keyvault as an option (although it's tackling the same situation, how to get the storage key and put it there in the first place), but if anything else fails, I can give it a go.
Was looking at [https://github.com/Azure/bicep/discussions/6173](https://github.com/Azure/bicep/discussions/6173) , but not sure if that'd work either.
Could you please advise on the best (and secure) approach here?

**Comment by eriqua** [comment]
Hi @sitoder, thanks for opening this discussion and sorry for the late reply. Discussions are currently not monitored promptly and are planned to be disabled in favor of the new issue template [Questions/Feedback].

I'm converting this discussion to an issue in order for the team to be able to triage it.

Please refer to issue #3141 going forward.
